src/helpers/stores/columns.js

```diff
@@ -10,6 +11,20 @@ const UI_COLUMNS_STORAGE_KEY = 'ui-columns';
 const UI_COLUMN_VALUES = Object.values(UI_COLUMN);
 const UI_COLUMN_IDS = UI_COLUMN_VALUES.map((column) => column.id);
 
+DEFAULT_COLUMNS.map((dcolumn) => {
+  UI_COLUMN_VALUES.map((column) => {
+    if (dcolumn.id === column.id) {
+      dcolumn.enabled = config.COLUMNS.includes(column.label);
+      if (dcolumn.enabled) {
+        let fromIndex = DEFAULT_COLUMNS.indexOf(dcolumn);
+        let toIndex = config.COLUMNS.indexOf(column.label);
+        DEFAULT_COLUMNS.splice(fromIndex, 1);
+        DEFAULT_COLUMNS.splice(toIndex, 0, dcolumn);
```

Comment:
It would be nicer not to mutate the original array, it causes a side effect that is unexpected for the developer (aka: I import `DEFAULT_COLUMNS`, and find an enabled key, while the code in `constant/columns.js` doesn't have an enabled key). So instead I would rather like to add this lookup to the `src/helpers/constants/columns.js` (`enabled: config.COLUMNS.includes('[label]')`) or to regenerate a list of columns here with the enabled key in place and to always use that value as the import value rather than the one from `constants/columns.js`.

If you're familiar with Typescript the changes you're doing here would not be possible because you're changing the type of the original array.

Reply:
This only mutates the order of `DEFAULT_COLUMNS`, its what `splice` does, you are obviously confusing it with a since reverted change...